tests/test_spacetime_utils.py

```diff
@@ -70,13 +70,27 @@ def f(x):
 
 
 @pytest.mark.parametrize("block_dim", range(5, 10))
-def test_noise_estimation(medium_random_matrix, block_dim):
-    """Test noise estimation."""
-    noise_map = estimate_noise(medium_random_matrix, block_dim)
-
-    real_std = np.nanstd(medium_random_matrix)
-    err = np.nanmean(noise_map - real_std)
-    assert err <= 0.1 * real_std
+def test_noise_estimation(block_dim):
+    """Test noise estimation.
+
+    The mean patch-wise standard deviation should be close to the overall
+    standard deviation.
+    """
+    for seed in range(10):
+        print(f"Seed: {seed}")
+        rng = np.random.RandomState(seed)
+        medium_random_matrix = rng.randn(200, 200, 100)
+        print(f"Mean of raw: {np.nanmean(medium_random_matrix)}")
+        print(f"Max of raw: {np.nanmax(medium_random_matrix)}")
+        print(f"Min of raw: {np.nanmin(medium_random_matrix)}")
+
+        noise_map = estimate_noise(medium_random_matrix, block_dim)
+        print(f"Mean of noise map: {np.nanmean(noise_map)}")
+        real_std = np.nanstd(medium_random_matrix)
+        print(f"SD of raw: {real_std}")
+        err = np.nanmean(noise_map - real_std)
+        print(f"Err: {err}")
+        assert err <= 0.1 * real_std
```

Comment:
I'm digging into the intermittent test failures with the noise maps. The noise map sometimes has a really massive value. I'm not sure why.

```
Seed: 0
Mean of raw: 0.0004815311392502157
Max of raw: 4.796465495404996
Min of raw: -5.031764123854
Mean of noise map: 134637028848.53595
SD of raw: 0.9994477899078287
Err: 134637028847.5365
```

Reply:
If the thermal noise volumes have values from -5 to 5, I can't see any way for the noise map to have values from 0 to 527766886400. I think there's a bug in the noise map estimation.

Reply:
I can't reproduce the error locally. I've tried with seeds going from 0 to 60 and block sizes from 5 to 10, but never get these massive noise map values.

Reply:
I'm thinking it must be something to do with the float precision.